pkg/cluster/cluster.go

```diff
@@ -1733,16 +1733,56 @@ func (c *Cluster) GetStatus() *ClusterStatus {
 
 // Switchover does a switchover (via Patroni) to a candidate pod
 func (c *Cluster) Switchover(curMaster *v1.Pod, candidate spec.NamespacedName) error {
-
 	var err error
 	c.logger.Debugf("switching over from %q to %q", curMaster.Name, candidate)
+
+	if !isInMaintenanceWindow(c.Spec.MaintenanceWindows) {
+		c.logger.Infof("postponing switchover, not in maintenance window")
+
+		var possibleSwitchover, schedule time.Time
+
+		now := time.Now().UTC()
+		for _, window := range c.Spec.MaintenanceWindows {
+			if window.Everyday {
+				possibleSwitchover = time.Date(now.Year(), now.Month(), now.Day(), window.StartTime.Hour(), window.StartTime.Minute(), 0, 0, time.UTC)
+				if now.After(possibleSwitchover) {
+					// we are already past the time for today, try tomorrow
+					day := now.AddDate(0, 0, 1)
+					possibleSwitchover = time.Date(day.Year(), day.Month(), day.Day(), window.StartTime.Hour(), window.StartTime.Minute(), 0, 0, time.UTC)
+				}
+			} else {
+				timeToday := time.Date(now.Year(), now.Month(), now.Day(), window.StartTime.Hour(), window.StartTime.Minute(), 0, 0, time.UTC)
+				// is it still possible today?
+				if now.Weekday() == window.Weekday {
+					if now.Before(timeToday) {
+						possibleSwitchover = timeToday
+					}
```

Comment:
we only need to act on `if now.Weekday() != window.Weekday`, no?

Reply:
why? what if it is 10am while the window is at 20:00-22:00 today